R/calc_views.R

```diff
@@ -135,6 +137,18 @@ set_view_lim <- function(views){
       usr.lim <- views[[as.numeric(i)]][['window']][[lim.name]][1:2]
       views[[as.numeric(i)]][['window']][[lim.name]] <- data.lim
       views[[as.numeric(i)]][['window']][[lim.name]][!is.na(usr.lim)] <- usr.lim[!is.na(usr.lim)]
+      
+      usr.axs <- data[[paste0(var, 'axs')]][[as.numeric(i)]]
+      view.index <- which(!names(views[[as.numeric(i)]]) %in% c('window', 'gsplot'))[which(!is.na(usr.axs))]
+      
+      if (any(!is.na(usr.axs)) && usr.axs[which(!is.na(usr.axs))] == 'o') {
+        buffer <- 0.04*views[[as.numeric(i)]][['window']][[lim.name]][[2]] #4% buffer based on upper limit
```

Comment:
It does work for reversed axes because the reversal happens after the window limits are set:

```r
gs <- gsplot()  %>%
    points(1:4, xlim=c(0,NA), xaxs='o')  %>%
    axis(side=1, reverse=TRUE)
gs
```

logged axes do not work (thanks for point that out)